### ISSUE TITLE
No longer export md_ns() and instead assign the namespace by default?

### ISSUE THREAD

**Issue by maelle** [ropensci/tinkr#48]

**Comment by maelle** [comment]
What's your take @zkamvar?

**Comment by zkamvar** [comment]
I remember thinking about this earlier and abandoning the idea initially because I didn't really understand how to do it best (you can see my struggles in https://github.com/carpentries/pegboard/blob/main/.notes/namespaces.Rmd)

But now that I think of it, there _is_ a way of doing this that will prevent the need for both `md_ns()` AND `copy_xml()` (maybe), but there are other caveats that we have to deal with. I'm going to write up a short explanation of what I've found and how it impacts {tinkr}, {pegboard}, and any of the other revdeps.

In short, this change will (probably) be a breaking change because the `md:node` prefix will (potentially) no longer work (but that depends on how we handle the situation).

**Comment by zkamvar** [comment]
So, I think the easy solution for us will be the following strategy:

1. strip the namespace immediately on read
2. on write, add the namespace back in as a default namespace

The downsides to this are:

1. all code that used the `md:node` prefix strategy will need to modify the code
2. anyone who exports the raw XML will be missing the crucial namespace, so we might want to provide a wrapper to re-add the default namespace (and no, a prefixed namespace will not work)

**Comment by maelle** [comment]
> all code that used the md:node prefix strategy will need to modify the code

Hopefully that's a breaking change for the best?

> anyone who exports the raw XML will be missing the crucial namespace, so we might want to provide a wrapper to re-add the default namespace (and no, a prefixed namespace will not work)

Why would someone export the raw XML? (genuine question)

**Comment by maelle** [comment]
Of course this is no longer valid after reading the vignette. :grin:

**Comment by zkamvar** [comment]
> Why would someone export the raw XML? (genuine question)

I mean, {tinkr} does some neat parsing with the rmarkdown chunk attributes and anchor links. People might want to export it into XML so they can work with it in a language they understand better (like JS or Python).
